robot/impl/resource_manager.go

```diff
@@ -184,15 +184,53 @@ func (manager *resourceManager) updateRemoteResourceNames(
 ) bool {
 	manager.logger.CDebugw(ctx, "updating remote resource names", "remote", remoteName, "recreateAllClients", recreateAllClients)
 	activeResourceNames := map[resource.Name]bool{}
-	newResources := rr.ResourceNames()
+	ms, err := rr.MachineStatus(ctx)
+
+	var newResources []resource.Status
+	switch {
+	case err == nil:
+		newResources = ms.Resources
+
+	// The connection to the remote is broken. In this case, we mark each resource node
+	// on this remote as disconnected but do not report any other changes.
+	case errors.Is(err, client.ErrDisconnected):
+		remoteGraph, err := manager.resources.SubGraphFrom(remoteName)
+		if err != nil {
+			manager.logger.Error(
+				"unable to lookup remote resources internally",
+				"remote", remoteName,
+				"error", err,
+			)
+			return true
```

Comment:
unclear what the behavior should be here - it might be time to just make `updateRemoteResourceNames` return an error and defer that decision the caller.